pkg/block/mem/adapter.go

```diff
@@ -24,62 +25,92 @@ var (
 	ErrNoPropertiesForKey = fmt.Errorf("no properties for key")
 )
 
+type partInfo struct {
+	data         []byte
+	lastModified time.Time
+	size         int64
+}
+
 type mpu struct {
-	id    string
-	parts map[int][]byte
+	id        string
+	objectKey string // the key (from getKey(obj))
+	initiated time.Time
+	parts     map[int]partInfo // part number -> partInfo
 }
 
-func newMPU() *mpu {
+func newMPU(objectKey string) *mpu {
 	uid := uuid.New()
 	uploadID := hex.EncodeToString(uid[:])
 	return &mpu{
-		id:    uploadID,
-		parts: make(map[int][]byte),
+		id:        uploadID,
+		objectKey: objectKey,
+		initiated: time.Now(),
+		parts:     make(map[int]partInfo),
 	}
 }
 
+// get returns the concatenated data of all parts in the mpu.
+// assumed that the caller had already acquired the lock.
```

Comment:
```suggestion
// Not thread safe: lock around calls.
```
mpu itself doesn't _have_ a lock, so there is no "the" lock.